src/external_metadata.py

```diff
@@ -119,6 +108,60 @@ def resize_image() -> str:
         return None
 
 
+# async def encode_icon_to_data_uri(icon_name: str) -> str:
+#     """
+#     Encode an image from a local file path or remote URL.
+#
+#     Returns a base64-encoded PNG data URI.
+#     """
+#     if isinstance(icon_name, MediaImage):
+#         icon_name = icon_name.url
+#
+#     if isinstance(icon_name, str) and icon_name.startswith("data:image"):
+#         _LOG.debug("Icon is already a data URI")
+#         return icon_name
+#
+#     _LOG.debug("Encoding icon to data URI: %s", icon_name)
+#     try:
+#         if _is_url(icon_name):
+#             async with httpx.AsyncClient() as client:
+#                 response = await client.get(icon_name, timeout=10)
+#                 response.raise_for_status()
+#                 img_bytes = BytesIO(response.content)
+#
+#             def encode_image() -> str:
+#                 img = Image.open(img_bytes)
+#                 img = img.convert("RGBA")
+#                 buffer = BytesIO()
+#                 img.save(buffer, format="PNG")
+#                 encoded = base64.b64encode(buffer.getvalue()).decode("utf-8")
+#                 return f"data:image/png;base64,{encoded}"
+#
+#             return await asyncio.to_thread(encode_image)
+#
+#         def load_and_encode() -> str:
+#             icon_path = _get_icon_path(icon_name)
+#             if not icon_path.exists():
+#                 raise FileNotFoundError(f"Icon not found: {icon_name}")
+#             with open(icon_path, "rb") as f:
+#                 img = Image.open(f)
+#                 img.load()
+#                 img = img.convert("RGBA")
+#                 buffer = BytesIO()
+#                 img.save(buffer, format="PNG")
+#                 encoded = base64.b64encode(buffer.getvalue()).decode("utf-8")
+#                 return f"data:image/png;base64,{encoded}"
+#
+#         return await asyncio.to_thread(load_and_encode)
+#
+#     except Exception as e:
+#         _LOG.warning("Failed to encode icon to base64 for %s: %s", icon_name, e)
+#         return ""
```

Comment:
Please remove commented out code blocks, unless they are used "real soon" and with a comment about the intention.